pyrobosim/pyrobosim/manipulation/grasping.py

```diff
@@ -0,0 +1,547 @@
+            [left_bottom_base, left_bottom_tip,
+                left_top_tip, left_top_base],
+            # Gripper base
+            [right_bottom_base, left_bottom_base,
+                left_top_base, right_top_base]
+        ]
+        ax.add_collection3d(
+            Poly3DCollection(gripper_verts, color=color, alpha=alpha)
+        )
+
+    def __repr__(self):
+        """ Printable string representation """
+        display_str = f"Grasp:\n\t{self.origin}\n"
+        display_str += f"\tFace: {self.face}\n\tDirection: {self.direction}\n"
+        if self.properties is not None:
+            display_str += "\t{self.properties}\n"
+        return display_str
+
+
+class ParallelGraspProperties:
+    """
+    Representation of a parallel-jaw gripper.
+
+        SIDE VIEW               TOP VIEW
+                  |            _________
+        _____     v           |              ^
+    ===|     |  height     ===|          max_width
+        -----     ^           |_________     v
+                  |
+                              |<-depth->|
+           depth_clearance -->||<--
+    """
+    def __init__(self, max_width, depth, height, width_clearance=0.0, depth_clearance=0.0):
+        """
+        Creates a parallel gripper grasp properties instance.
+
+        :param max_width: Maximum gripper opening width 
+        :type max_width: float
+        :param depth: Distance from end effector base to finger tips (or equivalent)
+        :type depth: float
+        :param height: Height of end effector finger tips (or equivalent)
+        :type height: float
+        :param width_clearance: Width clearance so grasps are not flush with gripper fingers
+        :type width_clearance: float, optional
+        :param depth_clearance: Depth clearance so grasps are not flush with gripper base
+        :type depth_clearance: float, optional
+        """
+        self.max_width = max_width
+        self.depth = depth
+        self.height = height
+        self.width_clearance = width_clearance
+        self.depth_clearance = depth_clearance
+
+    def __repr__(self):
+        """ Printable string representation """
+        display_str = "Parallel jaw gripper properties:\n"
+        display_str += f"\tMax width: {self.max_width}, depth: {self.depth}, height: {self.height}"
+        return display_str
+
+
+class GraspGenerator:
+    """
+    Generates grasps given object dimensions and pose relative to a robot.
+    """
+    def __init__(self, properties):
+        """
+        Creates a grasp generator instance given grasping properties.
+
+        :param properties: Grasping properties object
+        :type properties: :class:`pyrobosim.manipulation.grasping.ParallelGraspProperties`
+        """
+        self.properties = properties
+
+    def compute_robot_facing_rot(self, object_pose=Pose(), robot_pose=None):
+        """
+        Computes the rotation matrix to convert from nominal cuboid orientation to robot-facing orientation.
```

Comment:
I have trouble understanding this. Do you mean, it transforms the cuboid orientation to the robot frame?

Reply:
I do mean that! I can clarify